frontend/src/machines/billings/billingMachine.ts

```diff
@@ -0,0 +1,517 @@
+              target: "getMembersError",
+              actions: "updateContext",
+            },
+          },
+        },
+        getMembersError: {
+          on: {
+            REFTECH_MEMBERS: "loadingMembers",
+          },
+        },
+        getBillingDetailCondition: {
+          always: [
+            {
+              cond: "isCreatingForm",
+              target: "#billing.billingFormReady",
+            },
+            {
+              target: "getBillingDetailData",
+            },
+          ],
+        },
+        getBillingDetailData: {
+          on: {
+            FETCH_BILLING_DETAIL_SUCCES: {
+              target: "#billing.billingFormReady",
+              actions: "updateContext",
+            },
+            FETCH_BILLING_DETAIL_ERROR: {
+              target: "getBillingDetailDataError",
+              actions: "updateContext",
+            },
+          },
+        },
+        getBillingDetailDataError: {
+          on: {
+            REFTECH_BILLING_DETAIL: "getBillingDetailData",
+          },
+        },
+      },
+    },
+    billingFormReady: {
+      initial: "firstStep",
+      states: {
+        firstStep: {
+          on: {
+            NEXT_STEP: {
+              target: "secondStep",
+            },
+            UPDATE_FORM: {
+              actions: "updateContext",
+            },
+          },
+        },
+        secondStep: {
+          on: {
+            PREV_STEP: "firstStep",
+            UPDATE_FORM: {
+              actions: "updateContext",
+            },
+            SUBMIT_BILLING: "#billing.submitBilling",
+          },
+        },
+      },
+      on: {
+        CANCEL_FILL_FORM: "getBillingOK",
+      },
+    },
+    submitBilling: {
+      on: {
+        SUBMIT_BILLING_SUCCES: {
+          target: "submitBillingOK",
+        },
+        SUBMIT_BILLING_ERROR: {
+          target: "submitBillingError",
+          actions: "updateContext",
+        },
+      },
+    },
+    submitBillingOK: {
+      on: {
+        BACK_TO_BILLING_SCREEN: {
+          target: "loadingBilling",
+        },
+      },
+    },
+    submitBillingError: {
+      on: {
+        BACK_TO_SECOND_STEP_FORM: "billingFormReady.secondStep",
+        REFETCH_SUBMIT_BILLING: "submitBilling",
+      },
+    },
+  },
+}).withConfig({
+  actions: {
+    updateContext: assign((ctx, event) =>
+      match(event)
+        .with({ type: "FETCH_BILLING" }, () => ctx)
+        .with({ type: "ACTIVATE_BILLING_FORM" }, () => ctx)
+        .with({ type: "REFETCH_BILLING" }, () => ctx)
+        .with({ type: "REFTECH_MEMBERS" }, () => ctx)
+        .with({ type: "REFTECH_BILLING_DETAIL" }, () => ctx)
+        .with({ type: "REFETCH_SUBMIT_BILLING" }, () => ctx)
+        .with({ type: "NEXT_STEP" }, () => ctx)
+        .with({ type: "PREV_STEP" }, () => ctx)
+        .with({ type: "SUBMIT_BILLING" }, () => ctx)
+        .with({ type: "SUBMIT_BILLING_SUCCES" }, () => ctx)
+        .with({ type: "BACK_TO_BILLING_SCREEN" }, () => ctx)
```

Comment:
ini pas `BACK_TO_BILLING_SCREEN` kayaknya perlu reset beberapa property context nya jadi kayak gini : 

```
{
  ...ctx,
  formMode: "CREATE",
  members: [],
  membersError: null,
  billingDetail: undefined,
  billingDetailError: null,
  billingForm: undefined,
  submitBillingError: null,
}
```

Reply:
I missed that one, agree ka. Otherwise bakalan unexpected context value ketika mulai transitition state nya lagi.